src/node_task_runner.cc

```diff
@@ -1,35 +1,27 @@
 #include "node_task_runner.h"
 #include "util.h"
 
-#include <filesystem>
 #include <regex>  // NOLINT(build/c++11)
 
 namespace node::task_runner {
 
 #ifdef _WIN32
-static constexpr const char* bin_path = "\\node_modules\\.bin";
+static constexpr const char* env_var_separator = ";";
 #else
-static constexpr const char* bin_path = "/node_modules/.bin";
+static constexpr const char* env_var_separator = ":";
 #endif  // _WIN32
 
 ProcessRunner::ProcessRunner(std::shared_ptr<InitializationResultImpl> result,
-                             std::string_view package_json_path,
+                             const std::filesystem::path& package_json_path,
                              std::string_view script_name,
                              std::string_view command,
+                             std::string_view path_env_var,
                              const PositionalArgs& positional_args) {
   memset(&options_, 0, sizeof(uv_process_options_t));
 
-  // Get the current working directory.
-  char cwd[PATH_MAX_BYTES];
-  size_t cwd_size = PATH_MAX_BYTES;
-  CHECK_EQ(uv_cwd(cwd, &cwd_size), 0);
-  CHECK_GT(cwd_size, 0);
-
-#ifdef _WIN32
-  std::string current_bin_path = cwd + std::string(bin_path) + ";";
-#else
-  std::string current_bin_path = cwd + std::string(bin_path) + ":";
-#endif  // _WIN32
+  package_json_path_ = std::filesystem::path(package_json_path);
```

Comment:
We are inside the constructor, right? I would expect the code to use a member initializer list, which initializes class members before the constructor body runs. E.g. 

```C++
ProcessRunner::ProcessRunner(std::shared_ptr<InitializationResultImpl> result,
                              const std::filesystem::path& package_json_path,
                              std::string_view script_name,
                              std::string_view command,
                              std::string_view path_env_var,
                              const PositionalArgs& positional_args) : package_json_path_(package_json_path) {
```

And so forth.

I don't particular care and it was an issue prior to this PR. I don't think that the current code is bad even if I am correct... although, if I am correct, a langage purist would object.

There might be a small (non measurable) performance issue, although I am not certain.